sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/SynchronousMessageSubscriber.java

```diff
@@ -210,7 +211,8 @@ private void drainQueue() {
                 if (!isEmitted) {
                     // The only reason we can't emit was the downstream(s) were terminated hence nobody
                     // to receive the message.
-                    if (isPrefetchDisabled) {
+                    // In RECEIVE_AND_DELETE mode, we re-buffer the message because it is deleted in service side.
+                    if (isPrefetchDisabled && asyncClient.getReceiverOptions().getReceiveMode() == ServiceBusReceiveMode.PEEK_LOCK) {
                         // release is enabled only for no-prefetch scenario.
```

Comment:
Nit; Is it possible to consolidate all comments in one place (preferably here)? by updating current comment to something like -

"When Prefetch is disabled, for the receive mode that influences the delivery count (today, only PeekLock ReceiveMode), we try to release undelivered messages to adjust the delivery count on the broker"